generate_openj9_scc.sh

```diff
@@ -0,0 +1,295 @@
+        APT_INSTALL_CMD="DEBIAN_FRONTEND=noninteractive"
+        APT_INSTALL_CMD="${APT_INSTALL_CMD} apt install -y"
+
+        if [ "$INSTALL_GTK" == true ]; then
+            # install gtk+3.0
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} gtk+3.0"
+        fi
+
+        if [ "$INSTALL_XVFB" == true ]; then
+            # install xvfb
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} xvfb"
+        fi
+
+        eval "$APT_INSTALL_CMD"
+    fi
+}
+
+# function to download the applications
+function download_and_install_artifacts() {
+
+    # check for eclipse
+    if [ "$RUN_ECLIPSE" == true ]; then
+        # Creating a temporary directory for eclipse download
+        mkdir -p "$DOWNLOAD_PATH_ECLIPSE" "$INSTALL_PATH_ECLIPSE"
+
+
+        # Downloading eclipse
+        wget --directory-prefix="$DOWNLOAD_PATH_ECLIPSE"  http://www.mirrorservice.org/sites/download.eclipse.org/eclipseMirror/technology/epp/downloads/release/2020-06/M3/eclipse-java-2020-06-M3-linux-gtk-x86_64.tar.gz
+
+        # Extracting eclipse
+        tar -xvzf "$DOWNLOAD_PATH_ECLIPSE"/eclipse-java-2020-06-M3-linux-gtk-x86_64.tar.gz -C "$INSTALL_PATH_ECLIPSE"
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_ECLIPSE"
+    fi
+
+    # check for tomcat
+    if [ "$RUN_TOMCAT" == true ]; then
+        # Creating a temporary directory for tomcat download
+        mkdir -p "$DOWNLOAD_PATH_TOMCAT" "$INSTALL_PATH_TOMCAT"
+
+        # Downloading tomcat
+        wget --directory-prefix="$DOWNLOAD_PATH_TOMCAT" https://downloads.apache.org/tomcat/tomcat-9/v9.0.35/bin/apache-tomcat-9.0.35.tar.gz
+
+        # Extracting tomcat
+        tar -xvzf "$DOWNLOAD_PATH_TOMCAT"/apache-tomcat-9.0.35.tar.gz -C "$INSTALL_PATH_TOMCAT"
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_TOMCAT"
+    fi
+
+}
+
+# dry run to right size the cache
+function dry_run() {
+    # Creating base layer first instead of running sample programs to generate SCC directly
+    java -Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    # Pointing cache for sample program runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
+        # Spawning a virtual screen for GUI apps
+        Xvfb :1 -ac -screen 0 1024x768x8 &
+        # Saving the PID
+        XVFB_PID=$!
+        # Saving older display value
+        OLD_DISPLAY=$DISPLAY
+        # Setting DISPLAY to created screen
+        export DISPLAY=:1
+    fi
+
+    for ((i=0; i<SCC_GEN_RUNS_COUNT; i++))
+    do
+        if [ "$RUN_ECLIPSE" == true ]; then
+            run_eclipse_and_stop
+        fi
+        if [ "$RUN_TOMCAT" == true ]; then
+            run_tomcat_and_stop
+        fi
+    done
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
```

Comment:
Similarly, `stop_xvfb` here. You can do the `if` check inside the function

Reply:
created `check_and_stop_xvfb` to check `xvfb` installation and kill the screen